experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,290 @@
+# Oak Baremetal Communication Protocol Specification
+
+This document serves as a specification of the protocol used for communication
+between the Untrusted Loader and Trusted Runtime. The Trusted Runtime should be
+able to communicate with any implementation of the Untrusted Loader that
+implements this specification.
+
+In the specification below the Trusted Runtime will henceforth be referred to
+the as the "service". The Untrusted Loader will be referred to the as the
+"client".
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).
+
+## Layer Defintions
+
+### Method Layer
+
+Methods are exposed by the service. Each method has clearly defined parameters
+and return types. The Oak IDL is used to define which methods are available,
+their id, parameter and return types, as well the protcol used to encode
+parameters and return values to bytes.
+
+#### Invocation
+
+An invocation of a Method. Each invocation consists out of a request followed by
+a response.
+
+##### Initiating an invocation
+
+The client MAY initiate an invocation by sending a request.
+
+The service MUST NOT send requests.
+
+##### Handling an invocation
+
+The service MUST respond to each request with a single response message. It
+SHOULD respond to requests in first-in-first-out order, but MAY prioritize some
+requests. The service MAY send a response before the request has been received
+in its entirety. Sending a response concludes the invocation.
+
+The client MUST NOT send response.
+
+### Message Layer
```

Comment:
Before describing each layer individually, could you please add a small overview of all of the layers (with their encodings) that will be described later?